GQL framework (#212)
neo4j/docs-status-codes: modules/ROOT/pages/notifications/index.adoc

After a successful query execution, the Neo4j server sends notifications to provide advice on how to improve the query's quality or give additional information about the query execution.

> The driver receives these notifications and sends them to the client, which displays them to the user.

Author: I understand your point. We mention the word client in other places as well. For example, 
```
Client errors::
These errors are caused by the client and are usually related to the request itself.
The Neo4j status codes for client errors have the prefix `Neo.ClientError`.
```

 Will adding a clarification such as "client (e.g. Browser/Bloom/Cypher Shell)" make it better?

From version 5.23, Neo4j has a new GqlStatusObject API in addition to the existing Notification API (deprecated since 5.26).
The GqlStatusObject API provides information about the status of a Cypher query or command execution in compliance with the link:https://www.iso.org/standard/76120.html[ISO/IEC 39075:2024(en) - Information technology - Database languages - GQL Standard].

> This page describes both the GQL-status object and the Neo4j Notification frameworks, how they are structured, the objects they provide for notifications, and how to interpret them.

Author: I guess we could say:
```suggestion
This page describes the GQL-status object and the Neo4j-status object frameworks, their structure, the objects they provide for notifications, and how to interpret them.
```

For more information on the Bolt versions supported by different server versions, see the link:https://neo4j.com/docs/bolt/current/bolt-compatibility[Bolt Protocol documentation].

On the server side, notifications are part of the Result Core API.
A method called `getNotifications()`, which is deprecated since 5.26, returns a list of server-side notification objects.
These notifications are then sent to the driver as success Bolt message metadata.
On the driver side, notifications are part of the ResultSummary API, which has a method called `notifications()` that returns a list of driver-side Notification objects.

> The result of the `getCode()` or `code()` methods is known as the  Neo4j status code.

Reviewer: ```suggestion
The result of the `getCode()` or `code()` methods is known as the Neo4j status code.
```
